core/src/main/scala/org/bykn/bosatsu/SourceConverter.scala

```diff
@@ -348,6 +348,19 @@ final class SourceConverter(
             else RecursionKind.NonRecursive
           Expr.Let(boundName, lam, in, recursive = rec, decl)
         }
+      case IfElse(NonEmptyList((Matches(a, p), res), Nil), elseCase) =>
```

Comment:
It's better to make sure there are no names here and error than rewrite possibly with names and then have a less precise error condition.